### PR TITLE
PM-4954 Migrate SSO Component

### PR DESCRIPTION
**Pull request by KiruthigaManivannan**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [x] Other
```

## Objective

Migrate the SSO component to use the component library.

## Code changes

- **sso.component.html:** Added form group and bit form fields
- **sso.component.ts:** Added new form group , converted submit to arrow function
- **oss-routing.module.ts:** Added Routing for Anon layout wrapper component.

## Screenshots

https://github.com/bitwarden/clients/assets/162679756/48587419-a729-44f9-84c9-1c68c3ebf177

### CODE REVIEW

**Comment by KiruthigaManivannan** [comment]
@trmartin4  Does this SSO component changes looks good  ? Please advise

File: apps/web/src/app/auth/sso.component.ts [hunk 1]
```diff
@@ -82,7 +87,7 @@ export class SsoComponent extends BaseSsoComponent {
     this.route.queryParams.pipe(first()).subscribe(async (qParams) => {
       if (qParams.identifier != null) {
         // SSO Org Identifier in query params takes precedence over claimed domains
-        this.identifier = qParams.identifier;
+        this.formGroup.get("identifier").setValue(qParams.identifier);
```
Comment (JaredSnider-Bitwarden) [inline apps/web/src/app/auth/sso.component.ts:90]: Nit: please add an accessor for the identifier form control: 
```Typescript
  get identifierFormControl(): FormControl {
    return this.formGroup.get("identifier") as FormControl;
  }
  ```

Then you can simply use `this.identifier` to get the form control.

File: apps/web/src/app/auth/sso.component.ts [hunk 2]
```diff
@@ -131,13 +136,12 @@ export class SsoComponent extends BaseSsoComponent {
     }
   }
 
-  async submit() {
-    await this.ssoLoginService.setOrganizationSsoIdentifier(this.identifier);
+  submit = async () => {
+    this.identifier = this.formGroup.value.identifier;
+    await this.ssoLoginService.setOrganizationSsoIdentifier(this.formGroup.value.identifier);
```
Comment (JaredSnider-Bitwarden) [inline apps/web/src/app/auth/sso.component.ts:141]: You could simply pass `this.identifier` into `this.ssoLoginService.setOrganizationSsoIdentifier(...)`

File: apps/web/src/app/auth/sso.component.ts [hunk 3]
```diff
@@ -131,13 +136,12 @@ export class SsoComponent extends BaseSsoComponent {
...
     if (this.clientId === "browser") {
       document.cookie = `ssoHandOffMessage=${this.i18nService.t("ssoHandOff")};SameSite=strict`;
     }
-    // FIXME: Verify that this floating promise is intentional. If it is, add an explanatory comment and ensure there is proper error handling.
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    super.submit();
-  }
+    await Object.getPrototypeOf(this).submit.call(this);
```
Comment (JaredSnider-Bitwarden) [inline apps/web/src/app/auth/sso.component.ts:145]: Please replace this call with `await super.submit()`

File: apps/web/src/app/oss-routing.module.ts [hunk 4]
```diff
@@ -302,6 +298,21 @@ const routes: Routes = [
...
+        path: "sso",
+        component: SsoComponent,
+        canActivate: [unauthGuardFn()],
+        data: {
+          pageTitle: "enterpriseSingleSignOn",
+          titleId: "enterpriseSingleSignOn",
+        },
+      },
+    ],
+  },
```
Comment (JaredSnider-Bitwarden) [inline apps/web/src/app/oss-routing.module.ts:315]: We have just pushed up the new changes to the anon-layout component which allow for the proper env selector to be configured: https://components.bitwarden.com/?path=/docs/auth-anon-layout--docs#environment-selector. That will need to be configured as well to show the server selector.

**Review by JaredSnider-Bitwarden — COMMENTED** [review]
Nice work on this. A few comments below 👇

File: apps/web/src/app/auth/sso.component.ts (hunk 1 above)
Comment (JaredSnider-Bitwarden) [reply apps/web/src/app/auth/sso.component.ts]: @KiruthigaManivannan , I reworked the typescript to match what I was trying to show here, and I retested to ensure it worked.

**Review by JaredSnider-Bitwarden — APPROVED** [review]
LGTM!

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Pull request merged**